metaflow/plugins/events_decorator.py

```diff
@@ -96,47 +129,69 @@ def process_event_name(self, event):
                 "'parameters': {'alpha': 'beta'}})"
             )
 
-    def process_parameters(self, parameters):
+    def process_parameters(self, parameters, event_name):
+        """
+        Process the parameters for an event and return a dictionary of parameter mappings if
+        parameters was statically defined or a function if deploy-time trigger.
+
+        Parameters
+        ----------
+        Parameters : Union[callable, List[Union[str, Tuple[str, str]]], Dict[str, str]]
+            Parameters to process
+
+        event_name : Union[str, callable]
+            Name of the event
+
+        Returns
+        -------
+        Union[Dict[Union[str, callable], Union[str, callable]], Callable]
+            Processed parameters
+
+        Raises
+        ------
+        MetaflowException
+            If the parameters are not in the correct format
+        """
         new_param_values = {}
-        if isinstance(parameters, (list, tuple)):
+        if isinstance(parameters, list):
+            new_param_list = []
             for mapping in parameters:
                 if is_stringish(mapping):
+                    # param_name
                     new_param_values[mapping] = mapping
-                elif callable(mapping) and not isinstance(mapping, DeployTimeField):
-                    mapping = DeployTimeField(
-                        "parameter_val", str, None, mapping, False
-                    )
-                    new_param_values[mapping] = mapping
-                elif isinstance(mapping, (list, tuple)) and len(mapping) == 2:
-                    if callable(mapping[0]) and not isinstance(
-                        mapping[0], DeployTimeField
-                    ):
-                        mapping[0] = DeployTimeField(
-                            "parameter_val", str, None, mapping[0], False
-                        )
-                    if callable(mapping[1]) and not isinstance(
-                        mapping[1], DeployTimeField
-                    ):
-                        mapping[1] = DeployTimeField(
-                            "parameter_val", str, None, mapping[1], False
+                elif isinstance(mapping, tuple) and len(mapping) == 2:
+                    # (param_name, field_name)
+                    param_name, field_name = mapping
+                    if not is_stringish(param_name) or not is_stringish(field_name):
+                        raise MetaflowException(
+                            f"The *parameters* attribute for event {event_name} is invalid. "
```

Comment:
this error message does not seem to properly reflect the condition here. You are actually checking the the content of the tuple are also strings. That too, this is a list of strings or tuples.